SAMLPortal/Controllers/SetupController.cs

```diff
@@ -0,0 +1,268 @@
+					return Redirect("2");
+				}
+				catch (MySqlException ex)
+				{
+					Console.WriteLine(ex.Message);
+					ModelState.AddModelError(string.Empty, ex.Message);
+				}
+				catch (Exception)
+				{
+					ModelState.AddModelError(string.Empty, "An unknown error occured. Please try again.");
+				}
+			}
+
+			return View(model);
+		}
+
+		[HttpGet]
+		[Route("2")]
+		public IActionResult StepTwo()
+		{
+			var companyName = GlobalSettings.Get("CONFIG_CompanyName");
+			var appHost = GlobalSettings.Get("CONFIG_URL");
+
+			StepTwoModel model = new StepTwoModel();
+			model.CompanyName = companyName != null ? companyName : "";
+			model.AppHost = appHost != null ? appHost : Request.Host.ToString();
+
+			var countries = new CountryHelper().GetCountryData();
+			model.CountryList = new Microsoft.AspNetCore.Mvc.Rendering.SelectList(countries, "CountryShortCode", "CountryName");
+
+			return View(model);
+		}
+
+		[HttpPost]
+		[Route("2")]
+		public IActionResult StepTwo(StepTwoModel model)
+		{
+			var countries = new CountryHelper().GetCountryData();
+
+			if (ModelState.IsValid)
+			{
+				var country = countries.Where(c => c.CountryShortCode == model.CountryCode);
+				if (country.Any())
+				{
+					GlobalSettings.Store("CONFIG_CompanyName", model.CompanyName);
+					GlobalSettings.Store("CONFIG_CompanySubject", model.CompanyName);
+					GlobalSettings.Store("CONFIG_CompanyCountryCode", country.First().CountryShortCode);
+					GlobalSettings.Store("CONFIG_URL", model.AppHost);
+
+					Helpers.ReplaceEnvVariableInFile(GlobalSettings.Get("CONFIG_FILE"), "SP_CONFIG_SETUPASSISTANT_STEP", "3");
+					GlobalSettings.Store("CONFIG_SETUPASSISTANT_STEP", "3");
+
+					return Redirect("3");
+				}
+				else
+				{
+					ModelState.AddModelError(string.Empty, "Unknown country");
+				}
+
+			}
+
+			model.CountryList = new Microsoft.AspNetCore.Mvc.Rendering.SelectList(countries, "CountryShortCode", "CountryName");
+
+			return View(model);
+		}
+
+		[HttpGet]
+		[Route("3")]
+		public IActionResult StepThree()
+		{
+			var ldapHost = GlobalSettings.Get("LDAP_Host");
+			var ldapPort = GlobalSettings.GetInt("LDAP_Port");
+			var ldapSSL = GlobalSettings.Get("LDAP_SSL") != null ? bool.Parse(GlobalSettings.Get("LDAP_SSL")) : false;
+			var ldapBindDn = GlobalSettings.Get("LDAP_BindDN");
+			var ldapBindPassword = GlobalSettings.Get("LDAP_BindPass");
+
+			StepThreeModel model = new StepThreeModel();
+			model.Host = ldapHost != null ? ldapHost : "";
+			model.Port = ldapPort != null ? ((int)ldapPort) : 389;
+			model.SSL = ldapSSL;
+			model.BindDN = ldapBindDn != null ? ldapBindDn : "";
+			model.BindPassword = ldapBindPassword != null ? ldapBindPassword : "";
+
+			return View(model);
+		}
+
+		[HttpPost]
+		[Route("3")]
+		public IActionResult StepThree(StepThreeModel model)
+		{
+			if (ModelState.IsValid)
+			{
+				LdapConnection connection = new LdapConnection();
+				connection.SecureSocketLayer = model.SSL;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'model' before using it.](https://app.codacy.com/manual/zegorax/SAMLPortal/pullRequest?prid=5343548)